### PR TITLE
Add third-party BOM smoke module and centralise Java 8 quality gates

### PR DESCRIPTION
**Pull request by peter-lawrey**
Withdrawn as an umbrella. Retain the reduced compatibility suite in #323 and the independent Commons dependency-family/consumer work in #328. Neither requires merging this PR's unrelated parent, coverage or quality-policy changes.

An exact-source scanner control reproduces the missing ordinary dependencies and incorrect `software.chronicle` classification. Repository search found documentation recommending a future CI consumer, but no implemented consumer. No scanner cleanup is extracted here. Independently justified parent or coverage corrections should be reviewed separately with effective-model and execution evidence.

<details>
<summary>Original umbrella description for traceability</summary>

This PR introduces a dedicated `third-party-smoke` module to continuously validate the third-party BOM on Java 8 (and Java 21), tightens parent-level quality configuration, and refreshes documentation and project metadata to match the current build policy.

---

### Functional changes

* **New `third-party-smoke` module wired into the reactor**

  * Added `third-party-smoke` as a top-level module in the root `pom.xml` so it participates in the standard `mvn -q verify` run.
  * `third-party-smoke/pom.xml`:

    * Imports the `third-party-bom` via dependencyManagement and pulls in all third-party libraries under test.
    * Adds extensive test-scope dependencies grouped by category (observability, serialisation/parsing, Mongo drivers, HTTP servers/clients, logging, collections/utilities, databases, testing frameworks, OSGi/Pax, and additional peer libraries) so every BOM artefact is exercised.
    * Enforces **Java 8 bytecode** for all resolved dependencies with `maven-enforcer-plugin` and `extra-enforcer-rules`:

      * `enforceBytecodeVersion` with `maxJdkVersion` set to `1.8`.
      * Explicit exclusions for `org.hsqldb:hsqldb` and `org.mongodb:bson-record-codec`, documented as known non-Java-8 bytecode exceptions.
    * Enables **Checkstyle** (Sun checks, warning severity, fail on violation) and **SpotBugs** (Max effort, Medium threshold, fail on error) in the module’s `verify` phase to keep the smoke code clean.

* **Comprehensive class-loading smoke tests for BOM artefacts**

  * Added `ClassLoadingSmokeTest` which dynamically generates 80+ `@TestFactory` dynamic tests to ensure representative classes can be loaded for each third-party library.

    * Groups tests by category: observability, serialisation, Mongo, networking, logging, collections/utilities, databases, test frameworks, OSGi/Pax, and additional peer artefacts.
    * Uses a shared `ClassLoader`, `Class.forName`, ClassGraph scans, and resource lookups to prove jars are on the classpath and link correctly.
    * Handles known edge cases such as HSQLDB’s Java 11 classfiles by catching `UnsupportedClassVersionError` and treating it as acceptable on Java 8 when expected.

* **Minimal behaviour “smoke” tests per dependency family**

  * Added category-specific minimal usage tests to verify basic runtime behaviour beyond class loading:

    * `MinimalObservabilitySmokeTest`:

      * Registers and increments a Prometheus `Counter` and reads it back from a dedicated `CollectorRegistry`.
      * Builds an `OpenTelemetrySdk` with `SdkTracerProvider` and retrieves a tracer.
    * `MinimalSerializationSmokeTest`:

      * Uses JSONAssert for JSON comparison.
      * Parses simple YAML with SnakeYAML.
      * Round-trips a small POJO via XStream.
      * Generates a JSON Schema with Jackson jsonSchema.
      * Builds a JSON object with Jettison.
    * `MinimalNetworkingSmokeTest`:

      * Builds an Undertow server (binding to `localhost:0`) and immediately stops it.
      * Instantiates a Jetty `WebSocketClient`.
      * Configures a Netty `ServerBootstrap`.
      * Creates a Grizzly `FilterChainBuilder`.
      * Constructs and closes an OkHttp `MockWebServer`.
    * `MinimalLoggingSmokeTest`:

      * Creates and uses a Log4j2 logger via `LogManager`.
      * Creates and uses an SLF4J logger via `LoggerFactory`.
    * `MinimalUtilitiesSmokeTest`:

      * Exercises Trove maps, Commons Lang `StringUtils`, Commons CLI `Options`, JavaPoet code generation, Joda-Time `DateTime`, Commons Email `SimpleEmail`.
      * Compresses/decompresses data with Snappy.
      * Scans the classpath with ClassGraph.
      * Touches JNA (`Native.POINTER_SIZE`) and JNR (`Runtime.getSystemRuntime()`).
    * `MinimalDatabaseSmokeTest`:

      * Parses a MongoDB URI with `ConnectionString` and checks the database name.
      * Relies on the class-loading tests to validate HSQLDB driver presence.
    * `LegacyJUnit4SmokeTest`:

      * Simple JUnit 4 test to confirm the Vintage engine can still run legacy `@Test` methods.
  * Added `package-info.java` documenting the smoke test suite, mapping tests back to requirements and categories.

* **Third-party BOM fixes and alignment**

  * `third-party-bom/pom.xml`:

    * Corrected the **JUnit 5 migration support** dependency to use `junit-jupiter-migrationsupport` with `${junit.jupiter.version}` instead of `${junit.platform.provider.version}`, ensuring a valid 5.10.x coordinate for Java 8.
  * `chronicle-bom/pom.xml`:

    * Bumped `root-parent-pom` parent from `1.27ea4` to `1.27ea6`.
    * Updated multiple Chronicle component properties to newer `ea` and `-SNAPSHOT` versions (network, queue, zero, wire-enterprise, ring, FIX, services, logger, EFX, matching engine, platform, market data distributor, core, ITCH adapters, test framework, Posix, analytics, threads, bytes, values, wire, algorithms, map, JLBH, affinity, compiler, zero-allocation-hashing, etc.) to match current development heads.

* **Build behaviour changes**

  * `java-parent-pom/pom.xml`:

    * Switched parent from `root-parent-pom` `1.27ea7-SNAPSHOT` back to `1.27ea6` for consistency.
    * Updated `maven-surefire-plugin` configuration to set:

      * `argLine` to `${surefireArgLine} ${jvm.requiredArgs} -Djvm.coverage=true`, ensuring coverage-aware JVM flags are always present during tests.
  * `root-parent-pom/pom.xml`:

    * Added shared Checkstyle configuration properties:

      * `checkstyle.config.location` defaulting to `net/openhft/quality/checkstyle27/chronicle-baseline-checkstyle.xml`.
      * `checkstyle.violationSeverity` default `warning`.
      * `checkstyle.skip` default `false`.
    * Enabled stricter Javadoc checks by configuring `maven-javadoc-plugin` with `doclint` set to `all,-missing`, turning on all doclint rules except for missing Javadoc.
    * Removed the duplicated top-level `maven-checkstyle-plugin` configuration (Google checks + chronicle-quality-rules), leaving Checkstyle to be configured via the shared properties or module-level plugins.
    * Adjusted JaCoCo config so the `prepare-agent` execution is explicitly bound to the `test` phase.
    * Updated SCM `<tag>` from the older `root-parent-pom-1.27ea0` to `HEAD`, reflecting ongoing development.

---

### Non-functional changes (documentation, guidelines, housekeeping)

* **Chronicle-Core AGENTS guidance**

  * Added `AGENTS.md` tailored for Chronicle-Core:

    * Inherits base rules from the repository-level `AGENTS.md`.
    * Reinforces **British English** and **ISO-8859-1** policies, with guidance on avoiding smart quotes and unsafe Unicode and using textual forms (`>=`, `:alpha:`, `:yes:`) instead.
    * Clarifies Javadoc expectations: document behavioural contracts, edge cases, thread-safety, units, performance; avoid restating trivial getters/setters.
    * Captures build/test expectations (`mvn -q verify` from a clean checkout), commit and PR etiquette (≤72-character imperative subject, issue references, root-cause→fix→impact bodies).
    * Encourages real-time AsciiDoc updates, doc-first workflows, small coherent commits, and explicitly lists questions contributors should ask reviewers.
    * Introduces company-wide Nine-Box tagging taxonomy (FN, NF-P, NF-S, NF-O, TEST, DOC, OPS, UX, RISK) and a standard ADR template.
    * Adds AsciiDoc formatting guidelines (list indentation, emphasis/bold, section numbering with `:sectnums:`).

* **Global durable documentation for OpenHFT aggregation**

  * New `src/main/docs` tree:

    * `src/main/docs/README.adoc`:

      * Defines purpose and structure for durable docs (requirements, decisions, version issues).
      * States build expectations (`mvn -q verify` from root; `third-party-smoke` must pass on Java 8 and Java 21).
      * Documents that the smoke module enforces Java 8 bytecode via `enforceBytecodeVersion` and acts as the BOM gate.
      * Lists open topics (JUnit migration support alignment, HSQLDB artefacts, SLF4J bindings, multi-JDK matrix).
    * `src/main/docs/project-requirements.adoc`:

      * Captures aggregate requirements for the OpenHFT build, including:

        * `TEST-001` third-party BOM smoke coverage on Java 8 via `third-party-smoke`.
        * `NF-O-001` repeatable verification (`mvn -q verify` including smoke).
        * `NF-O-002` documentation currency for `src/main/docs`.
        * `NF-P-001` fast smoke runtime (no external IO).
        * `NF-O-003` Java 8 bytecode compatibility gate.
        * `NF-O-004` LTS runtime checks on Java 8 and Java 21.
    * `src/main/docs/decision-log.adoc`:

      * Adds ADR `ALL-TEST-001` to record the decision to create a third-party BOM smoke module as the Java 8 compatibility gate.
    * `src/main/docs/version-issues.adoc`:

      * Summarises version quirks and follow-ups:

        * JUnit migration support version correction to 5.10.0 and avoidance of JUnit 6 until a Java 17 baseline.
        * Bytecode guard configuration and known exclusions (HSQLDB, bson-record-codec).
        * HSQLDB Java 8 classifier and SLF4J binding behaviour.
        * Dual HSQLDB artefacts and future review.
        * Planned research on BOM entries, multi-JDK CI, Jetty/Pax/ProGuard alignment and SLF4J noise reduction.

* **Module-local docs for `third-party-smoke`**

  * `third-party-smoke/src/main/docs/README.adoc`:

    * Documents the module’s purpose, categories, how to run all tests or individual categories, and expected test counts.
    * Describes bytecode enforcement, exclusions, and open topics (hardcoded versions, SLF4J binding warnings, HSQLDB dual artefacts, multi-JDK matrix).
  * `third-party-smoke/src/main/docs/project-requirements.adoc`:

    * Defines module-level requirements (`SMOKE-TEST-*`, `SMOKE-NF-*`), mapping each to specific tests and configurations.
  * `third-party-smoke/src/main/docs/decision-log.adoc`:

    * Records smoke-specific ADRs (SMOKE-DEC-001..006) covering:

      * Dedicated module creation,
      * Separation of class-loading vs minimal usage tests,
      * Category-based test organisation,
      * Hardcoded version strategy,
      * Bytecode enforcement with exclusions,
      * Acceptance of multiple SLF4J bindings during tests.
  * `third-party-smoke/src/main/docs/version-issues.adoc`:

    * Lists detailed version quirks, bytecode exclusions, SLF4J binding rationale, and follow-up research tasks for the smoke module.

* **Asciidoc and content clean-ups**

  * Added common AsciiDoc header attributes (`:toc:`, `:lang: en-GB`, `:source-highlighter: rouge`) to multiple docs:

    * `README.adoc` (Chronicle Overview), `chronicle-bom/readme.adoc`, `docs/Anatomy.adoc`, `docs/Java-Version-Support.adoc`, `docs/Platform-Support.adoc`, `docs/Reducing-Garbage.adoc`, `docs/Version-Support.adoc`, `nexus-access.adoc`.
  * Improved wording, grammar, and British English spelling in marketing/overview docs:

    * Chronicle Overview: lower-cased “low-latency”, “microservices”, “off-the-shelf”, corrected “minimises”, “high-throughput”, etc.
    * Reducing garbage: clarified off-heap phrasing, fixed typos (“serialisation/deserialise”), improved headings and examples.
  * Tweaked the Graphviz snippet in `docs/Anatomy.adoc` to remove trailing spaces and normalise formatting.
  * Removed a trailing blank line from `Platform-Support.adoc` and `allstatus.sh`.

* **Metadata and parent alignment**

  * `nexus-access.adoc` and other docs now have consistent AsciiDoc headers.
  * `root-parent-pom` SCM tag moved to `HEAD` to reflect ongoing development in the parent.

</details>

### CODE REVIEW

File: third-party-smoke/pom.xml [hunk 1]
```diff
@@ -0,0 +1,622 @@
...
+                        <phase>validate</phase>
+                        <goals>
+                            <goal>enforce</goal>
+                        </goals>
+                        <configuration>
+                            <rules>
+                                <enforceBytecodeVersion>
+                                    <maxJdkVersion>1.8</maxJdkVersion>
+                                    <excludes>
+                                        <!-- Known non-Java-8 bytecode from upstream: see src/main/docs/version-issues.adoc -->
```
Comment (peter-lawrey) [inline third-party-smoke/pom.xml]: This needs to be addressed in the third-party-bom

Comment (peter-lawrey) [reply third-party-smoke/pom.xml]: These libraries support both versions so falsely show an issue

File: chronicle-bom/pom.xml [hunk 2]
```diff
@@ -32,21 +32,21 @@
         chronicle dependencyManagement section.
     </description>
     <properties>
-        <chronicle.network.version>2.27ea7</chronicle.network.version>
-        <chronicle.queue.version>5.27ea11</chronicle.queue.version>
-        <chronicle.zero.version>2.27ea1</chronicle.zero.version>
-        <chronicle.wire.enterprise.version>2.27ea1</chronicle.wire.enterprise.version>
+        <chronicle.network.version>2.27ea8-SNAPSHOT</chronicle.network.version>
```
Comment (tgd) [inline chronicle-bom/pom.xml:35]: We just need to back out the version changes from this PR

**Review by tgd — CHANGES_REQUESTED** [review]
